src/raftstore/coprocessor/dispatcher.rs

```diff
@@ -151,12 +152,15 @@ impl CoprocessorHost {
     pub fn pre_propose(&self, region: &Region, req: &mut RaftCmdRequest) -> Result<()> {
         if !req.has_admin_request() {
             let query = req.mut_requests();
-            try_loop_ob!(
+            let mut vec_query = mem::take(query).into();
+            let result = try_loop_ob!(
                 region,
                 &self.registry.query_observers,
                 pre_propose_query,
-                query
-            )
+                &mut vec_query,
+            );
+            *query = vec_query.into();
```

Comment:
can it hurt the performance? another clone?

Reply:
There is no extra clone, the `into`s only copy the `Vec` header, not the contents